### PR TITLE
Add My Korean Kitchen

### LINKED ISSUE #1093: Add My Korean Kitchen
I'll be implementing a scraper for the site "[My Korean Kitchen](https://mykoreankitchen.com/)".

Test case link:

- https://mykoreankitchen.com/korean-sweet-rice-with-dried-fruit-and-nuts/

Other examples:

- https://mykoreankitchen.com/donkatsu/
- https://mykoreankitchen.com/korean-curry-rice/

To be implemented by myself.

### PR DESCRIPTION
**Pull request by a1831319**
This PR adds support for recipes from My Korean Kitchen. There was full schema support for this site for just about everything except ingredient groups, which needed to be configured manually.

I originally just scraped the HTML data directly for the ingredient groups, but the content on the page does not directly match the schema data which results in a mismatch between ingredient names. The solution I found was to grab the schema names directly from `self.schema.ingredients()` by keeping track of ranges (for example, the first group has 7 ingredients so the first 7 ingredients in the array belong to that group, the second group has 6 ingredients so ingredients 8 to 13 belong to that group, etc.). Do let me know if there's a better way to implement this : )

Resolves #1093

### CODE REVIEW

**Review by mlduff — CHANGES_REQUESTED** [review]

File: recipe_scrapers/mykoreankitchen.py [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+                    ],
+                    (
+                        ingredient_group.find(
+                            "h4", "wprm-recipe-ingredient-group-name"
+                        ).text
+                        if ingredient_group.find(
+                            "h4", "wprm-recipe-ingredient-group-name"
+                        )
+                        is not None
+                        else "MAIN"
```
[the remark this answers is not in the document]
Comment (a1831319) [reply recipe_scrapers/mykoreankitchen.py:64]: Yeah, that's fair, defaulting to None probably holds better as well since that'd indicate no grouping.

**Review by mlduff — APPROVED** [review]

File: recipe_scrapers/mykoreankitchen.py [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    def ingredients(self):
+        return self.schema.ingredients()
+
+    def ingredient_groups(self):
+        ingredient_list = self.soup.find_all(
+            "div", {"class": "wprm-recipe-ingredient-group"}
+        )
+
+        ingredient_groups = []
+        ingredient_index_base = 0
```
Comment (jayaddison) [inline recipe_scrapers/mykoreankitchen.py:40]: I'd probably suggest renaming `ingredient_groups` to `results` here, allowing `ingredient_list` to be renamed to the more-precise `ingredient_groups`.

Comment (a1831319) [reply recipe_scrapers/mykoreankitchen.py:40]: Sounds good to me.

File: recipe_scrapers/mykoreankitchen.py [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    (
+                        ingredient_group.find(
+                            "h4", "wprm-recipe-ingredient-group-name"
+                        ).text
+                        if ingredient_group.find(
+                            "h4", "wprm-recipe-ingredient-group-name"
+                        )
+                        is not None
+                        else None
+                    ),
```
[the remark this answers is not in the document]
Comment (a1831319) [reply recipe_scrapers/mykoreankitchen.py:65]: That's fair, easier to read that way.

File: recipe_scrapers/mykoreankitchen.py [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            results.append(
+                IngredientGroup(
+                    [
+                        ingredient
+                        for ingredient in self.schema.ingredients()[
+                            ingredient_index_base:ingredient_count
+                        ]
+                    ],
+                    (purpose),
+                )
```
Comment (jayaddison) [inline recipe_scrapers/mykoreankitchen.py:67]:
```suggestion
            purpose_heading = ingredient_group.find(
                "h4", "wprm-recipe-ingredient-group-name"
            )

            results.append(
                IngredientGroup(
                    ingredients=[
                        ingredient
                        for ingredient in self.schema.ingredients()[
                            ingredient_index_base:ingredient_count
                        ]
                    ],
                    purpose=purpose_heading.text if purpose_heading else None,
                )
```

**Review by jayaddison — APPROVED** [review]
:+1: thanks @a1831319!  I'll plan to merge this soon (within the next few days)

**Pull request merged**